styx-common/src/main/java/com/spotify/styx/util/BasicWorkflowValidator.java

```diff
@@ -144,11 +137,97 @@ public BasicWorkflowValidator(DockerImageValidator dockerImageValidator) {
       }
     });
 
+    validateDockerExecConf(e, cfg);
+    validateFlyteExecConf(e, cfg);
+    validateConflictingExecConf(e, cfg);
+
     return e;
   }
 
   private static boolean validateServiceAccount(String serviceAccount) {
     var matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(serviceAccount);
     return matcher.matches();
   }
+
+  private void validateDockerExecConf(List<String> errors, WorkflowConfiguration config) {
+    validateDockerParam(errors, "dockerImage", config.dockerImage(),
+        config.dockerExecConf().flatMap(DockerExecConf::dockerImage),
+        this::validateDockerImage);
+    validateDockerParam(errors, "dockerArgs", config.dockerArgs(),
+        config.dockerExecConf().flatMap(DockerExecConf::dockerArgs),
+        this::validateDockerArgs);
+
+    // not verifying dockerTerminationLogging as it defaults to false
+    // So for
+    //       WorkflowConfiguration.builder()
+    //          .id("styx.TestEndpoint")
+    //          .commitSha(VALID_SHA)
+    //          .schedule(DAYS)
+    //          .serviceAccount("foo@bar.baz.quux")
+    //          .dockerExecConf(new DockerExecConfBuilder()
+    //              .dockerImage("busybox")
+    //              .dockerArgs(List.of("x", "y"))
+    //              .dockerTerminationLogging(true)
+    //              .build())
+    //          .build();
+    // will fail as default at upper level is false but inside dockerExecConf is set to true
+  }
+
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private <T> void validateDockerParam(List<String> errors, String paramName, Optional<T> valueA, Optional<T> valueB,
```

Comment:
I kind of feel that this abstraction creates unnecessary complication to the code